src/core/ddsc/src/dds_serdata_builtintopic.c

```diff
@@ -99,6 +106,53 @@ static struct ddsi_serdata_builtintopic *serdata_builtin_new(const struct ddsi_s
   return d;
 }
 
+struct format_address_arg {
+  char *buf;
+  size_t buf_pos;
+  size_t buf_size;
+  bool first;
+};
+
+static void format_address (const ddsi_xlocator_t *n, void *varg)
+{
+  struct format_address_arg *arg = varg;
+  char buf[DDSI_LOCSTRLEN];
+
+  if (!arg->buf) return;
+
+  ddsi_xlocator_to_string (buf, sizeof(buf), n);
+  const size_t nsize = strlen(buf) + (arg->first ? 0 : 1);
+
+  if (nsize > arg->buf_size - arg->buf_pos - 1) {
+    arg->buf_size += 4 * nsize;
+    arg->buf = ddsrt_realloc(arg->buf, arg->buf_size);
+
+    if (!arg->buf) return;
```

Comment:
That's a memory a leak (`realloc` is really annoying ... I fell into the same trap the other day and had to be rescued by @dpotman ...)